hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterStateIdContext.java

```diff
@@ -86,7 +86,7 @@ public void setResponseHeaderState(RpcResponseHeaderProto.Builder headerBuilder)
     }
     RouterFederatedStateProto.Builder builder = RouterFederatedStateProto.newBuilder();
     namespaceIdMap.forEach((k, v) -> {
-      if (v.get() != Long.MIN_VALUE) {
+      if ((v.get() != Long.MIN_VALUE) && RouterRpcClient.isNamespaceObserverReadEligible(k)) {
```

Comment:
Are you making all the other stuff static because of this?
I don't think this is clean.

Reply:
Thanks for reivew.  Yes, you are right. I will  try to initizalize this in RouterStateIdContext.java only.